pysat/tests/classes/cls_instrument_access.py

```diff
@@ -0,0 +1,840 @@
+        self.eval_successful_load(end_date=end_date)
+        return
+
+    def test_basic_instrument_load_by_date_with_extra_time(self):
+        """Ensure `.load(date=date)` only uses date portion of datetime."""
+
+        # put in a date that has more than year, month, day
+        self.testInst.load(date=(self.ref_time + dt.timedelta(minutes=71)))
+        self.eval_successful_load()
+        return
+
+    def test_basic_instrument_load_data(self):
+        """Test that correct day loads (checking down to the sec)."""
+
+        self.testInst.load(self.ref_time.year, self.ref_doy)
+        self.eval_successful_load()
+        return
+
+    def test_basic_instrument_load_leap_year(self):
+        """Test if the correct day is being loaded (Leap-Year)."""
+
+        self.ref_time = dt.datetime(2008, 12, 31)
+        self.ref_doy = 366
+        self.testInst.load(self.ref_time.year, self.ref_doy)
+        self.eval_successful_load()
+        return
+
+    @pytest.mark.parametrize("operator,ref_time",
+                             [('next', dt.datetime(2008, 1, 1)),
+                              ('prev', dt.datetime(2010, 12, 31))])
+    def test_file_load_default(self, operator, ref_time):
+        """Test if correct day loads by default when first invoking `.next`."""
+
+        getattr(self.testInst, operator)()
+
+        # Modify ref time since iterator changes load date.
+        self.ref_time = ref_time
+        self.eval_successful_load()
+        return
+
+    @pytest.mark.parametrize("operator", [('next'), ('prev')])
+    def test_file_load_bad_start_file(self, operator):
+        """Test Error for in new day when on a file not in iteration list."""
+
+        self.testInst.load(fname=self.testInst.files[12])
+
+        # Set new bounds that do not include this date.
+        self.testInst.bounds = (self.testInst.files[9], self.testInst.files[20],
+                                2, 1)
+        with pytest.raises(StopIteration) as err:
+            getattr(self.testInst, operator)()
+        estr = 'Unable to find loaded filename '
+        assert str(err).find(estr) >= 0
+
+        return
+
+    @pytest.mark.parametrize("operator", [('next'), ('prev')])
+    def test_file_load_bad_start_date(self, operator):
+        """Test that day iterators raise Error on bad start date."""
+
+        self.testInst.load(date=self.ref_time)
+
+        # Set new bounds that do not include this date.
+        self.testInst.bounds = (self.ref_time + dt.timedelta(days=1),
+                                self.ref_time + dt.timedelta(days=10),
+                                '2D', dt.timedelta(days=1))
+
+        with pytest.raises(StopIteration) as err:
+            getattr(self.testInst, operator)()
+        estr = 'Unable to find loaded date '
+        assert str(err).find(estr) >= 0
+
+        return
+
+    @pytest.mark.parametrize("operator", [('next'), ('prev')])
+    def test_file_load_empty_iteration(self, operator):
```

Comment:
I think this iteration test slipped through. Suggest moving to iteration file.

Reply:
moved